README.md

```diff
@@ -1,6 +1,67 @@
 # reproducibility
-Framework and tools for reproducibility testing of models
 
+## Overview
 
+Framework and reusable tools for reproducibility testing of models
 
-# **DISCLAIMER**: THIS IS A TEST REPOSITORY that is currently only for internal ACCESS-NRI testing
+## Usage
+
+As with most custom actions and reusable workflows, you call it across repositories with the following syntax:
+
+```yaml
+# ...
+call-reusable:
+  name: Call Reusable Workflow
+  uses: access-nri/reproducibility/.github/workflows/<workflow_name>@<commit>
+  with:
+    param: Test
+
+call-custom-action:
+  name: Call Custom Action
+  runs-on: ubuntu-latest
+  steps:
+    - name: Call Action
+      uses: access-nri/reproducibility/.github.actions/<action_name>@<commit>
+      with:
+        other-param: Test
+```
+
+## Reusable Workflows
+
+### Reproducibility CI Checks: `checks.yml`
+
+This workflow is a generic reusable workflow that runs reproducibility checks in a given deployment environment.
+
+> [!NOTE]
+> The `vars` and `secrets` within this workflow are inherited from the caller, NOT defined in this repository. Therefore, they must be defined by any workflow that calls this workflow, and any call to this workflow must have `secrets: inherit`.
+
+#### Inputs
+
+| Name | Type | Description | Required | Default | Example |
+| ---- | ---- | ----------- | -------- | ------- | ------- |
+| `model-name` | `string` | The name of the model to check for reproducibility | `true` | N/A | `"access-om2"` |
+| `config-tag` | `string` | A tag on an associated config branch to use for the reproducibility run | `true` | N/A | `"release-1deg_jra55_iaf-1.2"` |
+| `environment-name` | `string` | The name of a GitHub Deployment Environment that is inherited from the caller | `true` | N/A | `"Gadi"` |
+| `test-markers` | `string` (python-style expression) | A python expression of markers to pass to the reproducibility pytests `-m` flag. These pytests are defined in the caller | `true` | N/A | `"checksums and fast and not performance"` |
```

Comment:
Is

> "checksums and fast and not performance"

really a legitimate argument? Wow.

Reply:
Not _entirely_ legitimate...at the moment we just have the `fast` test marker, but wanted to show the type of expression that is required.

Reply:
I was surprised you can use what effectively looks like natural language.

Reply:
Thats why we love python :D It's actually a python expression that the `pytest -m` flag ingests.